File: shared/src/main/java/ru/privatenull/pnauth/limbo/LimboControl.java
package ru.privatenull.pnauth.limbo;

import java.util.UUID;
import java.util.Optional;

public interface LimboControl {
    int apiVersion();

    boolean isPlayerConnected(UUID playerId);

    LimboControlResult addBossBar(UUID playerId, UUID barId, String title, float progress,
                                  LimboBossBarColor color, LimboBossBarOverlay overlay);

    LimboControlResult updateBossBarProgress(UUID playerId, UUID barId, float progress);

    LimboControlResult updateBossBarTitle(UUID playerId, UUID barId, String title);

    LimboControlResult removeBossBar(UUID playerId, UUID barId);

    LimboControlResult showDialog(UUID playerId, String dialogJson);

    LimboControlResult clearDialog(UUID playerId);

    Optional<LimboDialogEvent> pollDialogEvent();
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/limbo/LimboControl.kt
package ru.privatenull.pnauth.limbo

import ru.privatenull.pnauth.display.BossBarColor
import ru.privatenull.pnauth.display.BossBarOverlay
import java.util.Optional
import java.util.UUID

interface LimboControl {
    fun apiVersion(): Int

    fun isPlayerConnected(playerId: UUID): Boolean

    fun addBossBar(
        playerId: UUID, barId: UUID, title: String, progress: Float,
        color: BossBarColor, overlay: BossBarOverlay
    ): LimboControlResult

    fun updateBossBarProgress(playerId: UUID, barId: UUID, progress: Float): LimboControlResult

    fun updateBossBarTitle(playerId: UUID, barId: UUID, title: String): LimboControlResult

    fun removeBossBar(playerId: UUID, barId: UUID): LimboControlResult

    fun showDialog(playerId: UUID, dialogJson: String): LimboControlResult

    fun clearDialog(playerId: UUID): LimboControlResult

    fun pollDialogEvent(): Optional<LimboDialogEvent>
}
